### PR TITLE
Updated sdk.py to support local install using ANDROID_HOME

### PR DESCRIPTION
**Pull request by JCash**
This change allows developers to use a local install of Android Studio for building the engine.
It means the developer doesn't have to run the package script, or take special care of SDK+NDK versions, and they can skip the `install_sdk` step.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: README_ANDROID.md [hunk 1]
```diff
@@ -4,6 +4,25 @@
 
 ### Android SDK/NDK
 
+When building the engine, the build system tries to find the sdk in this order:
+
+* `$DYNAMO_HOME/ext/SDKs/android-sdk` -  This is mainly for the Defold team and their CI
+* `$ANDROID_HOME` - Set this path to point to your android studio sdk installation (e.g. `ANDROID_HOME=~/Library/Android/sdk` is the default installation on macOS)
```
Comment (JCash) [inline README_ANDROID.md:10]: Currently only tested on macOS. SHould work the same on Linux I think.
On Windows, we should double check that Android Studio creates an ANDROID_HOME environment variable.

File: README_BUILD.md [hunk 2]
```diff
@@ -6,16 +6,28 @@ Defold uses the Python based build system [Waf](https://waf.io/). Most of the in
...
+
+* Windows - Visual Studio + clang++
+* macOS/iOS - XCode
+* Android - Android Studio
+* Linux - clang++
+* Consoles - The vendor specific sdks
+
+In the future, we want to support HTML5 as well, for easier setup for all platforms.
+
+If you have these tools installed, you can skip the `./scripts/build.py install_sdk` step altogether.
```
Comment (JCash) [inline README_BUILD.md:23]: The intent is to try to make this super clear to potential contributors.

File: build_tools/sdk.py [hunk 3]
```diff
@@ -154,6 +157,42 @@ def get_local_darwin_sdk_version(platform):
...
+## **********************************************************************************************
+## Android
+
+
+def get_android_local_sdk_path():
+    path = os.environ.get('ANDROID_HOME', None)
+
+    if path is None:
+        if sys.platform == 'darwin':
+            path = os.path.expanduser('~/Library/android/sdk')
```
Comment (JCash) [inline build_tools/sdk.py:169]: Not sure yet what the default folders are for Windows/Linux.
Doesn't matter too much, as ANDROID_HOME is used first.

File: build_tools/sdk.py [hunk 4]
```diff
@@ -154,6 +157,42 @@ def get_local_darwin_sdk_version(platform):
...
+
+    if path and os.path.exists(path):
+        return path
+
+    return None
+
+def _get_latest_version_from_folders(path):
+    dirs = [ x for x in os.listdir(path)]
+    dirs.sort(key=lambda x: tuple(int(token) for token in x.split('.')))
+    return dirs[0]
```
Comment (JCash) [inline build_tools/sdk.py:179]: If you install multiple versions, we'll always pick the latest.

File: build_tools/sdk.py [hunk 5]
```diff
@@ -398,10 +441,20 @@ def _get_defold_sdk_info(sdkfolder, platform):
...
 
+    elif platform in ('armv7-android', 'arm64-android'):
+        info['version'] = ANDROID_BUILD_TOOLS_VERSION
+        info['sdk'] = os.path.join(sdkfolder, "android-sdk")
+        info['ndk'] = os.path.join(sdkfolder, "android-ndk-r%s" % ANDROID_NDK_VERSION)
+        info['build_tools'] = os.path.join(info['sdk'], "build-tools" , ANDROID_BUILD_TOOLS_VERSION)
+        if platform == 'arm64-android':
+            info['api'] = ANDROID_64_NDK_API_VERSION
+        else:
+            info['api'] = ANDROID_NDK_API_VERSION
```
Comment (JCash) [inline build_tools/sdk.py:456]: We still want to control the actual supported API target

File: build_tools/waf_dynamo.py [hunk 6]
```diff
@@ -1649,7 +1642,8 @@ def detect(conf):
         conf.env['AR']       = '%s/%s-ar' % (bintools, tool_name)
         conf.env['RANLIB']   = '%s/%s-ranlib' % (bintools, tool_name)
         conf.env['LD']       = '%s/lld' % (bintools)
-        conf.env['DX']       = '%s/android-sdk/build-tools/%s/dx' % (ANDROID_ROOT, ANDROID_BUILD_TOOLS_VERSION)
+
+        conf.find_program('d8', var='D8', mandatory = True, path_list=[sdkinfo['build_tools']])
```
Comment (JCash) [inline build_tools/waf_dynamo.py:1646]: Making it a "find_program" makes sure we get a nice error if the `d8` (and in effect, the sdk) is missing.

**Review by AGulev — DISMISSED** [review]
🔥

**Review by AGulev — DISMISSED** [review]

**Pull request merged**